### PR TITLE
Add deprecation annotations

### PR DESCRIPTION
**Pull request by opokornyy**
**What this PR does / why we need it**:

This PR adds deprecation annotation to the tasks that will be deprecated and removed in future versions.

**Release note**:
```
None
```

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -38,6 +38,8 @@ Visit [RBAC permissions for running the tasks](docs/tasks-rbac-permissions.md) i
 
 ## Usage
 
+#### copy-template, modify-vm-template and create-vm-from-template tasks are  deprecated and will be removed in future versions.
```
Comment (ksimon1) [inline README.md:41]: please remove the extra space between `are  deprecated `

**Review by ksimon1 — CHANGES_REQUESTED** [review]
please use the https://www.conventionalcommits.org/en/v1.0.0/ for commits

File: README.md (hunk 1 above)
Comment (opokornyy) [reply README.md]: Done

File: README.md [hunk 2]
```diff
@@ -38,6 +38,8 @@ Visit [RBAC permissions for running the tasks](docs/tasks-rbac-permissions.md) i
 
 ## Usage
 
+#### copy-template, modify-vm-template and create-vm-from-template tasks are deprecated and will be removed in future versions.
```
Comment (0xFelix) [inline README.md:41]: Can we name a replacement yet? Or what's the plan going forward?

Comment (ksimon1) [reply README.md:41]: The plan is to deprecate template based tasks and replace them with create-vm-from-manifest with few enhancements related to instance types

File: templates/modify-vm-template/manifests/modify-vm-template.yaml [hunk 3]
```diff
@@ -3,6 +3,7 @@ apiVersion: tekton.dev/v1beta1
 kind: ClusterTask
 metadata:
   annotations:
+    task.kubevirt.io/deprecated: "true"
```
Comment (ksimon1) [inline templates/modify-vm-template/manifests/modify-vm-template.yaml:6]: can you please change the annotation to:
```
tekton.dev/deprecated: "true"
```

File: README.md (hunk 2 above)
Comment (0xFelix) [reply README.md:41]: Can we add this to the readme to give more context?

**Pull request merged**